tests/func-tests/virtual_machines_test.go

```diff
@@ -21,27 +24,52 @@ var _ = Describe("Virtual Machines", func() {
 	client, err := kubecli.GetKubevirtClient()
 	testscore.PanicOnError(err)
 
+	var (
+		workloadsNode      *corev1.Node
+		checkNodePlacement = false
+	)
+	workloadsNodes, err := client.CoreV1().Nodes().List(k8smetav1.ListOptions{
+		LabelSelector: "node.kubernetes.io/hco-test-node-type==workloads",
+	})
+
+	if err == nil && workloadsNodes != nil && len(workloadsNodes.Items) > 0 {
+		checkNodePlacement = true
+		workloadsNode = &workloadsNodes.Items[0]
+
+		fmt.Fprintf(GinkgoWriter, "Found Workloads Node. Node name: %s; node labels:\n", workloadsNode.Name)
+		w := json.NewEncoder(GinkgoWriter)
+		w.SetIndent("", "  ")
+
+		w.Encode(workloadsNode.Labels)
+	}
+
 	BeforeEach(func() {
 		tests.BeforeEach()
 	})
 
 	Context("vmi testing", func() {
-		It("should create verify and delete a vmi", func() {
-			vmi := testscore.NewRandomVMI()
-			vmiName := vmi.Name
-			Eventually(func() error {
-				_, err := client.VirtualMachineInstance(testscore.NamespaceTestDefault).Create(vmi)
-				return err
-			}, timeout, pollingInterval).Should(Not(HaveOccurred()), "failed to create a vmi")
-			Eventually(func() bool {
-				vmi, err = client.VirtualMachineInstance(testscore.NamespaceTestDefault).Get(vmiName, &k8smetav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				return vmi.Status.Phase == kubevirtv1.Running
-			}, timeout, pollingInterval).Should(BeTrue(), "failed to get the vmi Running")
-			Eventually(func() error {
-				err := client.VirtualMachineInstance(testscore.NamespaceTestDefault).Delete(vmiName, &k8smetav1.DeleteOptions{})
-				return err
-			}, timeout, pollingInterval).Should(Not(HaveOccurred()), "failed to delete a vmi")
-		})
+		for i := 0; i < 20; i++ {
+			It(fmt.Sprintf("should create, verify and delete a vmi; run #%d", i), func() {
+				vmi := testscore.NewRandomVMI()
+				vmiName := vmi.Name
+				Eventually(func() error {
+					_, err := client.VirtualMachineInstance(testscore.NamespaceTestDefault).Create(vmi)
+					return err
+				}, timeout, pollingInterval).Should(Not(HaveOccurred()), "failed to create a vmi")
+				Eventually(func() bool {
+					vmi, err = client.VirtualMachineInstance(testscore.NamespaceTestDefault).Get(vmiName, &k8smetav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					if vmi.Status.Phase == kubevirtv1.Running && checkNodePlacement {
```

Comment:
doesn't it obsolete the condition in line 66?

Reply:
Do you mean something like that?
```golang
Eventually(func() bool {
	vmi, err = client.VirtualMachineInstance(testscore.NamespaceTestDefault).Get(vmiName, &k8smetav1.GetOptions{})
	Expect(err).ToNot(HaveOccurred())
	if vmi.Status.Phase == kubevirtv1.Running {
		if checkNodePlacement {
			Expect(vmi.Labels["kubevirt.io/nodeName"]).Should(Equal(workloadsNode.Name))
			fmt.Fprintf(GinkgoWriter, "The VMI is running on the right node: %s\n", workloadsNode.Name)
		}
		return true
	}
	return false
}
```

Reply:
yes.
first ensure the VMI in question is running, and then check that it run on the expected node dedicated for workloads.
instead of checking twice that it is running.